dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -718,6 +727,45 @@ private void checkToolTokens(User authUser, Long userId, String organization) {
         return userDAO.findAll();
     }
 
+    // TODO: Do equivalent of below, but for Google users.
+    @GET
+    @Timed
+    @UnitOfWork
+    @RolesAllowed("admin")
+    @Path("/updateUserMetadataToGetIds")
+    @Operation(operationId = "updateUserMetadataToGetIds", description = "Update metadata of all GitHub users and returns list of Dockstore users who could not be updated.", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @ApiOperation(value = "Update metadata of all GitHub users and returns list of Dockstore users who could not be updated.", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, notes = "Admin only.", response = User.class, responseContainer = "List")
+    public List<User> updateUserMetadataToGetIds(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user")@Auth User user) {
+        List<Token> gitHubTokens = tokenDAO.findAllGitHubTokens();
+        List<User> usersNotUpdatedWithToken = new ArrayList<>();
+        List<User> usersNotUpdatedWithTokenOrUsername = new ArrayList<>();
+
+        for (Token t : gitHubTokens) {
+            User currentUser = userDAO.findById(t.getUserId());
+            if (currentUser != null) {
+                try {
+                    GitHubSourceCodeRepo gitHubSourceCodeRepo = (GitHubSourceCodeRepo)SourceCodeRepoFactory.createSourceCodeRepo(t);
+                    gitHubSourceCodeRepo.syncUserMetadataFromGitHub(currentUser);
```

Comment:
Do you need to check the return value of this method? It returns false if it fails. You do check it below.